lib/cchain/provider/abci.go

```diff
@@ -138,15 +139,119 @@ func newABCIFetchFunc(cl atypes.QueryClient) fetchFunc {
 		})
 		if err != nil {
 			incQueryErr(endpoint)
-			return nil, errors.Wrap(err, "abci query approved-from")
+			return nil, errors.Wrap(err, "abci query attestations-from")
 		}
 
-		atts, err := atypes.AttestationsFromProto(resp.Attestations)
+		if len(resp.Attestations) != 0 {
```

Comment:
What the probability of findings attestations at the latest height? Why do we first try and avoid searching by querying latest height, and if we don't find anything, then we start searching? We could also just always first search.

Reply:
We need metrics that show this optimisation is worth the extra query.

Reply:
Agreed. My instinct is that when things are working normally (relayer is keeping sync well), we'll mostly be querying at the latest height. But you're right, metrics are needed.